### PR TITLE
dojson: cantook json record transformation

### PR DESCRIPTION
**Pull request by reropag**
* NEW Creates json transformation rules.

Signed-off-by: Gianni Pante <gianni.pante@rero.ch>

./run-tests.sh

* to display the result of the cantookjson transformation as invenio JSON:
pipenv run dojson -d pjson -i <cantook_sample.json> do cantookjson

* to display the invenio marc21 generated from the cantookjson transformation:
pipenv run dojson -d pjson -i <cantook_sample.json> do cantookjson do tomarc21

### CODE REVIEW

File: tests/unit/test_dojson.py [hunk 1]
```diff
@@ -0,0 +1,379 @@
+# -*- coding: utf-8 -*-
```
Comment (rerowep) [inline tests/unit/test_dojson.py]: Better to use `conftest.py`
Where are the `pjson` dump and `to_marc21`transformation for better testing ?

Comment (reropag) [reply tests/unit/test_dojson.py]: it is back again

File: rero_ebooks/dojson/json/model.py [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+        'function_of_entity': 'Publication'
+    })
+    return None
+
+
+@cantook_json.over('physical_description', 'page_count')
+@utils.for_each_value
+@utils.filter_values
+def page_count(self, key, value):
+    """Title statement."""
```
Comment (jma) [inline rero_ebooks/dojson/json/model.py]: please change the comment

File: rero_ebooks/dojson/json/model.py [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+
+        return utils.GroupableOrderedDict(res)
+
+
+cantook_json = Underdo()
+"""rero_ebook Format for Cantook Data."""
+
+
+@cantook_json.over('system_control_number', 'id')
+def id(self, key, value):
```
Comment (jma) [inline rero_ebooks/dojson/json/model.py]: should be system_control_number

File: rero_ebooks/dojson/json/model.py [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+        result['type_of_personal_name_entry_element'] = 'Surname'
+    if value.get('nature') == 'author':
+        result['relator_code'] = 'aut'
+    elif value.get('nature') == 'translated_by':
+        result['relator_code'] = 'trl'
+    return result
+
+
+@cantook_json.over('electronic_location_and_access', 'cover|flipbook|link')
+def cover_flipbook_link(self, key, value):
```
Comment (jma) [inline rero_ebooks/dojson/json/model.py]: should be electronic_location_and_access

File: tests/unit/test_dojson.py [hunk 5]
```diff
@@ -0,0 +1,474 @@
...
+    """Test cover, flipbook, link tranformation.
+
+       The difference between this test and test_json_cover_flipbook_link_media
+       is the order of the input data. In this test the media data arrive
+       first and so, in the result, the first element in the list
+       'electronic_location_and_access' is the dict containing the key
+       'electronic_format_type' (instead of the last element in the
+       test: test_json_cover_flipbook_link_media).
+
+       Both test are need to have a 100 % test coverage.
```
Comment (jma) [inline tests/unit/test_dojson.py]: Both casses is used to test all parts of the implementation.

**Pull request merged**